src/views/darkroom.c

```diff
@@ -2282,6 +2300,11 @@ void connect_key_accels(dt_view_t *self)
   dt_accel_connect_view(self, "increase brush opacity", closure);
   closure = g_cclosure_new(G_CALLBACK(_brush_opacity_down_callback), (gpointer)self->data, NULL);
   dt_accel_connect_view(self, "decrease brush opacity", closure);
+
+  // undo/redo
+  closure = g_cclosure_new(G_CALLBACK(_darkroom_undo_callback), (gpointer)self, NULL);
```

Comment:
I'm pretty sure that you should just add second `_darkroom_undo_callback`, and just make those call appropriate `dt_undo_do_undo`/`dt_undo_do_redo`